### PR TITLE
Hotseat hide

### LINKED ISSUE #2573: minimap is revealed when turn is passed
**Describe the bug**
Minimap is not hidden when turn is passed to another player in hotseat mode. Thus, player can see opponent's location.

**Did it work earlier?**
IDK

**Screenshots**
Снимок экрана 2023-08-13 в 01 32 32

**Version**
 - Version: 1.3

**Additional context**

### PR DESCRIPTION
**Pull request by Laserlicht**
fixes #2573

What do you think?

Uses dimming to hide complete screen.

image

### CODE REVIEW

**Comment by Nordsoft91** [comment]
looks like workaround. Shouldn't be FOW instead?

**Comment by Laserlicht** [comment]
Yes, a workaround. Until we have a better solution. :)

File: client/adventureMap/AdventureMapInterface.cpp [hunk 1]
```diff
@@ -172,6 +173,8 @@ void AdventureMapInterface::dim(Canvas & to)
 		if (!casted && !window->isPopupWindow())
 		{
 			int backgroundDimLevel = settings["adventure"]["backgroundDimLevel"].Integer();
+			if(isHotseatMessage)
+				backgroundDimLevel = 255;
```
Comment (Nordsoft91) [inline client/adventureMap/AdventureMapInterface.cpp:177]: why not moving backgroundDimLevel to class fields and set it to 255/settings from onHotseatWaitStarted/onPlayerTurnStarted methods?

Comment (Laserlicht) [reply client/adventureMap/AdventureMapInterface.cpp:177]: Would also be possible. But I think it's more understandable like this? Or not?

Comment (Nordsoft91) [reply client/adventureMap/AdventureMapInterface.cpp:177]: surely it's personal preference. Extra variable which is used only to control another variable from another method a bit confusing. Controlling dimming directly in onHotseatWaitStarted states clearly, that you want to dim screen on hotseat message

Comment (Laserlicht) [reply client/adventureMap/AdventureMapInterface.cpp]: Okay, convinced. Are fewer lines. :)

**Comment by IvanSavenko** [comment]
> looks like workaround. Shouldn't be FOW instead?

Strictly speaking, fow cover should be trivial - add one check to `MapRendererBaseContext::isVisible`

However it won't cover town list / hero list / minimap / etc. So might not be a bad solution meanwhile.

**Comment by Laserlicht** [comment]
> However it won't cover town list / hero list / minimap / etc. So might not be a bad solution meanwhile.

This was the intention of this PR.

**Pull request merged**